src/api/app/controllers/concerns/authenticator.rb

```diff
@@ -0,0 +1,130 @@
+module Authenticator
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :extract_user
+    before_action :check_user_state
+    before_action :check_anonymous_access
+    before_action :track_user_login
+  end
+
+  def extract_user
+    user = if ::Configuration.proxy_auth_mode_enabled?
+             find_or_create_proxy_user
+           elsif request.session[:login] # Webui Session Auth
+             User.find_by!(login: request.session[:login])
+           elsif authorization_headers.present? # API Basic Auth
+             User.find_with_credentials(basic_auth[:login], basic_auth[:password])
+           end
+
+    User.session = user || User.find_nobody!
+    Rails.logger.debug { "User.session set to #{User.possibly_nobody.login}" }
```

Comment:
```suggestion
    Rails.logger.debug { "User.session set to #{User.session.login}" }
```
Just in the line above, `User.session` could be assigned to the nobody user.

We could also remove the `Rails.logger.debug` line.

Reply:
Ops, forget the suggestion. `User#self.session` returns `nil` if the user is `nobody`. :thinking: